cpp/ql/src/Best Practices/Likely Errors/CommaBeforeMisleadingIndentation.ql

```diff
@@ -0,0 +1,33 @@
+/**
+ * @name Comma before misleading indentation
+ * @description The expressions before and after the comma operator can be misread because of an unusual difference in start columns.
+ * @kind problem
+ * @id cpp/comma-before-misleading-indentation
+ * @problem.severity recommendation
+ * @tags maintainability
+ *       readability
+ */
+
+import cpp
+import semmle.code.cpp.commons.Exclusions
+
+Expr normalizeExpr(Expr e) {
```

Comment:
Can you add a QLDoc to this predicate?

Reply:
Ah! The QLDoc makes it a lot more clear what's going on. Thank you!

Could this predicate be written more declaratively as:
```ql
Expr normalizeExpr(Expr e) {
  result =
    min(Expr child, int column |
      child.getParentWithConversions*() = e and column = child.getLocation().getStartColumn()
    |
      child order by column
    )
}
```
?

Reply:
cc @d10c on ^.

Reply:
Trying it out, looks promising.